### ISSUE TITLE
Upload to non OSF Storage

### ISSUE THREAD

**Issue by ethanwhite** [dib-lab/osf-cli#96]
Is it currently possible to upload a file to a storage option that isn't `OSF Storage`?

I'm using the figshare add-on and would like to be able to upload to its storage area, but can't figure out how to indicate this. `osf ls` shows the storage area and any files in it:

```
$ osf ls
figshare/data.csv
osfstorage/data2.csv
osfstorage/data.csv
```

But if I try to upload using:

```
osf upload data3.csv figshare/data3.csv
```

I end up creating a `figshare` directory in `osfstorage`:

```
ethan@gandalf:~/osftest$ osf ls
figshare/data.csv
osfstorage/figshare/data3.csv
osfstorage/data2.csv
osfstorage/data.csv
```

**Comment by nkmeyers** [comment]
try https://osf.io/y9jdt/wiki/Typical Workflow/ Focus on 3&4 and use figshare instead of OSF Storage for the provider in step 4 . . .  suggestion from @brianjgeiger . . . not sure how this'll work in OSFCLI yet tho?

**Comment by betatim** [comment]
@ethanwhite yes and no. You can use other storage providers (after hooking them up via osf.io) but currently #84 means you can only use OSF and GitHub.

**Comment by betatim** [comment]
As always PRs welcome. It could be as easy as adding the name "figshare" here: https://github.com/dib-lab/osf-cli/blob/master/osfclient/utils.py#L9

**Comment by ethanwhite** [comment]
Closing since this particular issue was addressed in #100 and #84 covers the more general issue.
